server/storage/sqldb.go

```diff
@@ -68,19 +99,25 @@ func (db *SQLStorage) UpdateCurrent(gun data.GUN, update MetaUpdate) error {
 
 	if err := func() error {
 		// write new TUFFile entry
-		if err = translateOldVersionError(tx.Create(&TUFFile{
+		tufFile := TUFFile{
 			Gun:     gun.String(),
 			Role:    update.Role.String(),
 			Version: update.Version,
 			SHA256:  hexChecksum,
 			Data:    update.Data,
-		}).Error); err != nil {
+		}
+		if err = translateOldVersionError(tx.Create(&tufFile).Error); err != nil {
+			return err
+		}
+
+		// The Channel is added separately from the TUFFile record to prevent gorm from issuing an Update
+		if err = tx.Model(&tufFile).Association("Channels").Append(update.Channels).Error; err != nil {
```

Comment:
Apologies, the `Append` implementation in the Gorm code is kinda opaque - I think this will not error if a relationship already exists, since I think it eventually does an `INSERT ... WHERE NOT EXISTS ...`, but would it be possible to add a test case for if the metadata already exists in one of the channels specified, but not in the others?

Just to make sure this behavior is consistent should we ever chose to switch ORMs, or upgrade, etc.